test/common/ssl/context_impl_test.cc

```diff
@@ -295,6 +298,52 @@ TEST_F(SslContextImplTest, TestNoCert) {
   EXPECT_TRUE(context->getCertChainInformation().empty());
 }
 
+// Multiple RSA certificates are rejected.
+TEST_F(SslContextImplTest, AtMostOneRsaCert) {
+  envoy::api::v2::auth::DownstreamTlsContext tls_context;
+  NiceMock<Server::Configuration::MockTransportSocketFactoryContext> factory_context;
+  const std::string tls_context_yaml = R"EOF(
+  common_tls_context:
+    tls_certificates:
+    - certificate_chain:
+        filename: "{{ test_rundir }}/test/common/ssl/test_data/selfsigned_cert.pem"
+      private_key:
+        filename: "{{ test_rundir }}/test/common/ssl/test_data/selfsigned_key.pem"
+    - certificate_chain:
+        filename: "{{ test_rundir }}/test/common/ssl/test_data/selfsigned2_cert.pem"
```

Comment:
Could you re-use existing certs (`selfsigned` & `selfsigned_rsa_1024`* and `selfsigned_ecdsa_p256` and `selfsigned_ecdsa_p384`) instead of adding another set? We have a bit too many test certs already, and we could get rid of the `OUTPUT_PREFIX` hack in `certs.sh`, then.

*either copy it from #5318 or wait until it's merged (later today?)

Reply:
I think if we did that, it might work, because we check for duplicate certs of a given type before validating the cert, but this is just an implementation detail which could potentially change so let's stick with this for now.

Long term, I think the right direction to go here is to have some custom Go/Python script for programatically generating the certs (since we can't rely on the openssl CLI in any Bazel automation), so we don't need to check in so many certs.

Reply:
> Long term, I think the right direction to go here is to have some custom Go/Python script for programatically generating the certs (since we can't rely on the openssl CLI in any Bazel automation), so we don't need to check in so many certs.

We used to do openssl in genrule and that broke once in OSX, but I think we may enable that again by using bssl binary?